src/build.d

```diff
@@ -1480,6 +1486,75 @@ class Dependency
         }
     }
 
+     // Metadata required for parallel execution
+    private
+    {
+        shared uint builtDeps; /// amount of dependencies already built
+        bool visited; /// whether this dependency was already visited
+        Dependency[] requiredBy; /// dependencies relying on this one
+    }
+
+    /**
+    Builds all supplied targets using the global taskPool to execute buildRecursive
+    for every leaf dependency.
+
+    Params:
+        targets = dependencies to build
+    **/
+    static void build(Dependency[] targets)
+    {
+        import core.atomic: atomicLoad, atomicOp, atomicStore;
+
+        //// Recursively appends all dependencies without further dependencies to roots
+        static void findRoots(Dependency cur, ref Dependency[] roots)
+        {
+            if (cur.visited) return;
+            cur.visited = true;
+
+            if (cur.deps.empty)
+            {
+                roots ~= cur;
+            }
+            else
+            {
+                foreach (dep; cur.deps)
+                {
+                    dep.requiredBy ~= cur;
+                    findRoots(dep, roots);
+                }
+            }
+        }
+
+        /// Recursively builds cur and other dependencies requiring it
```

Comment:
NOTE: this is just discussion about `build.d`, not for this particular PR

This comment is ambiguous/confusing because we've been overloading the term "dependency".  Since the class is called `Dependency`, we can refer to any build object as a `dependency`.  However, the term "Dependency" also describes a relationship between 2 objects (i.e. A is a dependency of B).  This comment demonstrates how this can be confusing because you've said:
```
Recursively builds cur and other dependencies requiring it
```

If we interpret the word `dependencies` as objects with a `dependency` relationship to `cur`, then we can reword it like this

```
Recursively builds cur and "the components it requires" that require it
```

So it's saying that this function builds `cur` and it's circular dependencies.  I was able to gather that in this case 
`dependencies` was only describing `Dependency` objects rather that objects with a "dependency" relationship to `cur`, but in general the meaning is ambiguous.

I think the best way to avoid this confusion would be to rename the `Dependency` class to something else like `Rule` or `BuildRule` or something.  If that were the case, then your comment would read:
```
Recursively builds cur and other rules that require it
```

cc @wilzbach